src/posix/platform/netif.cpp

```diff
@@ -1563,6 +1564,79 @@ static void processNetifInfoEvent(otInstance *aInstance, struct rt_msghdr *rtm)
 
 #endif
 
+#if defined(__linux__)
+
+#define ERR_RTA(errmsg, requestPayloadLength) \
+    ((struct rtattr *)((char *)(errmsg)) + NLMSG_ALIGN(sizeof(struct nlmsgerr)) + NLMSG_ALIGN(requestPayloadLength))
+
+// The format of NLMSG_ERROR is described below:
+//
+// ----------------------------------------------
+// | struct nlmsghdr - response header          |
+// ----------------------------------------------------------------
+// |    int error                               |                 |
+// ---------------------------------------------| struct nlmsgerr |
+// | struct nlmsghdr - original request header  |                 |
+// ----------------------------------------------------------------
+// | ** optionally (1) payload of the request   |
+// ----------------------------------------------
+// | ** optionally (2) extended ACK attrs       |
+// ----------------------------------------------
+//
+static void HandleNetlinkResponse(struct nlmsghdr *msg)
+{
+    const struct nlmsgerr *err = reinterpret_cast<const nlmsgerr *>(NLMSG_DATA(msg));
+    size_t                 rtaLength;
+    size_t                 requestPayloadLength = 0;
+    uint32_t               requestSeq           = err->msg.nlmsg_seq;
+
+    if (msg->nlmsg_len < NLMSG_LENGTH(sizeof(struct nlmsgerr)))
```

Comment:
If this is truncated, is it safe to use the `requestSeq`?

Reply:
Thanks for catching this! I moved all initialization to after the truncation check to be consistent (including the pointer cast which is okay before truncation check).